libr/core/disasm.c

```diff
@@ -1685,6 +1685,17 @@ static void ds_show_functions(RDisasmState *ds) {
 	ds->stackptr = core->anal->stackptr;
 	RAnalFcnVarsCache vars_cache;
 	r_anal_fcn_vars_cache_init (core->anal, &vars_cache, f);
+
+	int o_varsum = ds->show_varsum;
+	if (!ds->show_varsum) {
+		int numvars = vars_cache.bvars->length + vars_cache.rvars->length + vars_cache.svars->length;
+		if (numvars > ds->l) {
```

Comment:
if ds->l s the amount of lines of the screen.. maybe we want to do osome -5 or -10 for padding

Reply:
also i think we should only do this when in interactive mode, for visual

Reply:
i agree with both, 30 mins

Reply:
better 10 i guess for Vpp

Reply:
done.